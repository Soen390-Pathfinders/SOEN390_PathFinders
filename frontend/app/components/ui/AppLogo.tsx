//App logo component
//Renders the Campus Pilot logo
import { View, Image, StyleSheet } from "react-native";
import useTheme from "../../hooks/useTheme";

export default function AppLogo() {
  const { theme, toggleTheme } = useTheme(); // Use the custom hook
  return (
    <View>
      <Image
        style={styles.logo}
        source={require("../../../assets/images/logo.png")}
        tintColor={theme === "dark" ? "#000" : "#fff"} // Invert the color based on theme
      />
    </View>
  );
}
const styles = StyleSheet.create({
  logo: {
    width: 50,
    height: 50,
  },
});
