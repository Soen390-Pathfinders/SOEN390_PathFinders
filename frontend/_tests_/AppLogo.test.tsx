import React from 'react';
import { render } from '@testing-library/react-native';
import AppLogo from '../app/components/ui/AppLogo';

// Mock the useTheme hook
jest.mock('../app/hooks/useTheme', () => ({
  __esModule: true,
  default: jest.fn(),
}));

import useTheme from '../app/hooks/useTheme';

describe('AppLogo Component', () => {
    it('renders correctly with light theme', () => {
      (useTheme as jest.Mock).mockReturnValue({ theme: 'light', toggleTheme: jest.fn() });
  
      const { toJSON } = render(<AppLogo />);
      expect(toJSON()).toMatchSnapshot();
    });
  
    it('renders correctly with dark theme', () => {
      (useTheme as jest.Mock).mockReturnValue({ theme: 'dark', toggleTheme: jest.fn() });
  
      const { toJSON } = render(<AppLogo />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
  