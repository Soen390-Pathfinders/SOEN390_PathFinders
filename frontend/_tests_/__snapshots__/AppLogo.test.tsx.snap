// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLogo Component renders correctly with dark theme 1`] = `
<View>
  <Image
    source={
      {
        "testUri": "../../../assets/images/logo.png",
      }
    }
    style={
      {
        "height": 50,
        "width": 50,
      }
    }
    tintColor="#000"
  />
</View>
`;

exports[`AppLogo Component renders correctly with light theme 1`] = `
<View>
  <Image
    source={
      {
        "testUri": "../../../assets/images/logo.png",
      }
    }
    style={
      {
        "height": 50,
        "width": 50,
      }
    }
    tintColor="#fff"
  />
</View>
`;
